modules/govuk_jenkins/manifests/jobs/deploy_app_downstream.pp

```diff
@@ -0,0 +1,30 @@
+# == Class: govuk_jenkins::jobs::deploy_app_downstream
+#
+# Creates jobs that enables kicking off deploys in a downstream environment.
+#
+# === Parameters
+#
+# [*applications*]
+#   Array of applications that are available to kick off downstream deploys.
+#
+# [*jenkins_downstream_api_user*]
+#   The username used to authenticate with the downstream deployment Jenkins.
+#
+# [*jenkins_downstream_api_password*]
+#   The password used to authenticate with the downstream deployment Jenkins.
+#
+# [*deploy_url*]
+#   The URL of the downstream Jenkins.
+#
+class govuk_jenkins::jobs::deploy_app_downstream (
```

Comment:
I agree that's another way of looking at this job. To me, it's equivalent to manually going and starting the `Deploy_App` job, so it made sense to use a related name. We could say `Trigger_Deploy_App_Downstream`, although this job does (/will) do a fair bit more than that. Perhaps we can wait to see if it confuses people?

Reply:
What else does/will this job do?

Reply:
At the moment, it does a couple of extra things: checking the TAG is acceptable to deploy downstream. In the future (slight deviation from the original plan):

- It will honour the "deploy_freeze" from the release app
- It will run specific Smokey tests, after waiting for the app to restart*

*This was originally going to be in Deploy_App, but I realised it would either cause confusion (to make the job support both the old and the new), or generate too much work (to convert all apps to run Smokey in the new way).

Does that help?

Reply:
Yup - sounds like its still ultimately "triggering" a downstream environment to deploy the app, rather than actually deploying. I think something like `Trigger_Deploy_App_Downstream` more indicative of its primary purpose.

Reply:
Let me know if you aren't going to change it, I'll go ahead and approve the dependent PRs.

Reply:
I'd like to leave it as-is for now. Mainly because of the effort of re-opening all those secrets files. I wouldn't mind if we changed it in future.